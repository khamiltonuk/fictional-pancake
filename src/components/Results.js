import React from "react";

const Controls = props => {
  return (
    <div>
      <h2 class="screenreader">Results</h2>
      <ul class="results">
        <li>Win: {`${props.win}`}</li>
        <li>Draw:{`${props.draw}`}</li>
        <li>Lose: {`${props.lose}`}</li>
      </ul>
    </div>
  );
};

export default Controls;
