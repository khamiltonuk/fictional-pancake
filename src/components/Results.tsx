import React from "react";
import styled from "styled-components";
import type { ResultsT } from "../types";

const Title = styled.h2`
  position: absolute;
  text-indent: -999em;
`;

const Result = styled.li`
  display: inline-block;
  min-width: 70px;
  margin: 2px;
  color: #da727e;
`;

const ResultsList = styled.ul`
  margin: 0;
  padding: 0;
  width: 100%;
  top: 60px;
  position: absolute;
  left: 0;
  list-style: none;
  text-align: center;
`;

const Results = ({ win, draw, lose }: ResultsT) => {
  return (
    <>
      <Title>Results</Title>
      <ResultsList>
        <Result>Win: {`${win}`}</Result>
        <Result>
          Draw:
          {`${draw}`}
        </Result>
        <Result>Lose: {`${lose}`}</Result>
      </ResultsList>
    </>
  );
};

export default Results;
