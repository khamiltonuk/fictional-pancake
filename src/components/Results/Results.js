// @flow
import React, { Fragment } from "react";
import styled from "styled-components";
import { type ResultsT } from "../../types";

const Title = styled.h2`
  position: absolute;
  text-indent: -999em;
`;

const Result = styled.li`
  display: inline-block;
  min-width: 70px;
  margin: 2px;
  color: #da727e;
`;

const Results = styled.ul`
  margin: 0;
  padding: 0;
  width: 100%;
  top: 60px;
  position: absolute;
  left: 0;
  list-style: none;
  text-align: center;
`;

const Controls = ({ win, draw, lose }: ResultsT) => {
  return (
    <Fragment>
      <Title>Results</Title>
      <Results>
        <Result>Win: {`${win}`}</Result>
        <Result>
          Draw:
          {`${draw}`}
        </Result>
        <Result>Lose: {`${lose}`}</Result>
      </Results>
    </Fragment>
  );
};

export default Controls;
