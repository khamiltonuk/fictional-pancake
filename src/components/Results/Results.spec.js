import React from "react";
import { shallow } from "enzyme";
import Results from "./Results";
import toJson from "enzyme-to-json";

describe("Results", () => {
  test("should match the snapshot", () => {
    const props = {
      win: 0,
      draw: 0,
      lose: 0
    };
    const wrapper = shallow(<Results {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
