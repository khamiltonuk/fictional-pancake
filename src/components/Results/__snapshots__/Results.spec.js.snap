// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should match the snapshot 1`] = `
<React.Fragment>
  <styled.h2>
    Results
  </styled.h2>
  <styled.ul>
    <styled.li>
      Win: 
      0
    </styled.li>
    <styled.li>
      Draw:
      0
    </styled.li>
    <styled.li>
      Lose: 
      0
    </styled.li>
  </styled.ul>
</React.Fragment>
`;
